extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/OidcClient.java

```diff
@@ -0,0 +1,21 @@
+package io.quarkus.oidc.client;
+
+import java.io.Closeable;
+
+import io.smallrye.mutiny.Uni;
+
+/**
+ * Token grant client
+ */
+public interface OidcClient extends Closeable {
+
+    /**
+     * Returns the grant tokens
+     */
+    Uni<GrantTokens> getTokens();
+
+    /**
+     * Refreshes the grant tokens
+     */
+    Uni<GrantTokens> refreshTokens(String refreshToken);
```

Comment:
Maybe a no-arg `refreshTokens` would help to avoid boilerplate code to get refresh token and then call this method?

Reply:
But `OidcClientImpl` is not keeping the tokens itself. I guess in a good number of cases the users won't use this client directly but via the filter which is hiding these expiry and refresh tokens checks